plugins/hubspot/src/hooks/useSearchParams.ts

```diff
@@ -0,0 +1,37 @@
+import { useSearch, useLocation } from "wouter"
+import { useMemo, useCallback } from "react"
+
+export const useSearchParams = <T extends Record<string, string>>() => {
+    const searchString = useSearch()
+    const [, setLocation] = useLocation()
+
+    // Parse current search params
+    const params = useMemo(() => {
+        const searchParams = new URLSearchParams(searchString)
+        return Object.fromEntries(searchParams.entries()) as T
```

Comment:
Maybe just straight up return `searchParams`? And use `.get` later? Less processing, and safer (types are lying here a bit).

Reply:
Yeah, that would be better.